tests/test_processing_common.py

```diff
@@ -335,3 +342,64 @@ def test_overlapping_text_kwargs_handling(self):
                 padding="max_length",
                 text_kwargs={"padding": "do_not_pad"},
             )
+
+
+class MyProcessor(ProcessorMixin):
+    attributes = ["image_processor", "tokenizer"]
+    image_processor_class = "CLIPImageProcessor"
+    tokenizer_class = ("CLIPTokenizer", "CLIPTokenizerFast")
+    optional_call_args = ["optional_1", "optional_2"]
+
+    def __init__(self, image_processor=None, tokenizer=None, processor_attr_1=1, processor_attr_2=True):
+        super().__init__(image_processor, tokenizer)
+
+        self.processor_attr_1 = processor_attr_1
+        self.processor_attr_2 = processor_attr_2
+
+
+@require_tokenizers
+@require_vision
+class ProcessorTest(unittest.TestCase):
+    processor_class = MyProcessor
+
+    def prepare_processor_dict(self):
+        return {"processor_attr_1": 1, "processor_attr_2": False}
+
+    def get_processor(self):
+        image_processor = CLIPImageProcessor.from_pretrained("openai/clip-vit-large-patch14")
+        tokenizer = CLIPTokenizerFast.from_pretrained("openai/clip-vit-large-patch14")
+        processor = MyProcessor(image_processor, tokenizer, **self.prepare_processor_dict())
+
+        return processor
+
+    def test_processor_to_json_string(self):
+        processor = self.get_processor()
+        obj = json.loads(processor.to_json_string())
+        for key, value in self.prepare_processor_dict().items():
+            self.assertEqual(obj[key], value)
+            self.assertEqual(getattr(processor, key, None), value)
+
+    def test_processor_from_and_save_pretrained(self):
+        processor_first = self.get_processor()
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            saved_file = processor_first.save_pretrained(tmpdirname)[0]
+            check_json_file_has_correct_format(saved_file)
+            processor_second = self.processor_class.from_pretrained(tmpdirname)
+
+        self.assertEqual(processor_second.to_dict(), processor_first.to_dict())
+
+    def test_prepare_and_validate_optional_call_args(self):
+        processor = self.get_processor()
+        # test all optional call args are given
+        optional_call_args = processor.prepare_and_validate_optional_call_args("optional_1", "optional_2")
+        self.assertEqual(optional_call_args, {"optional_1": "optional_1", "optional_2": "optional_2"})
+        # test only one optional call arg is given
+        optional_call_args = processor.prepare_and_validate_optional_call_args("optional_1")
+        self.assertEqual(optional_call_args, {"optional_1": "optional_1"})
+        # test no optional call arg is given
+        optional_call_args = processor.prepare_and_validate_optional_call_args()
+        self.assertEqual(optional_call_args, {})
+        # test too many optional call args are given
+        with self.assertRaises(ValueError):
+            processor.prepare_and_validate_optional_call_args("optional_1", "optional_2", "optional_3")
```

Comment:
I realize this was removed in a previous PR, because it only applied to one model I suppose? Can we take the tests here and move them to the tester class above instead, wdyt?

Reply:
You are right, it probably existed to test the test Mixin before any processor with the mixin wasmerged but I guess it is not useful anymore.
I moved `test_prepare_and_validate_optional_call_args` to the tester class above (the other tests were already there).